packages/ckb/src/utils/cell-dep.ts

```diff
@@ -0,0 +1,95 @@
+import axios from 'axios';
+import { getBtcTimeLockDep, getRgbppLockDep, getUniqueTypeDep, getXudtDep } from '../constants';
+
+interface CellDepsObject {
+  rgbpp: {
+    mainnet: CKBComponents.CellDep;
+    testnet: CKBComponents.CellDep;
+  };
+  btcTime: {
+    mainnet: CKBComponents.CellDep;
+    testnet: CKBComponents.CellDep;
+  };
+  xudt: {
+    testnet: CKBComponents.CellDep;
+  };
+  unique: {
+    testnet: CKBComponents.CellDep;
+  };
+}
+const GITHUB_CELL_DEPS_JSON_URL =
+  'https://raw.githubusercontent.com/ckb-cell/typeid-contract-cell-deps/main/deployment/cell-deps.json';
+
+const fetchCellDepsJson = async () => {
+  try {
+    const response = await axios.get(GITHUB_CELL_DEPS_JSON_URL);
+    return response.data as CellDepsObject;
+  } catch (error) {
+    // console.error('Error fetching cell deps:', error);
+  }
+};
+
+export interface CellDepsSelected {
+  rgbpp?: boolean;
+  btcTime?: boolean;
+  xudt?: boolean;
+  unique?: boolean;
+}
+
+export const fetchTypeIdCellDeps = async (
+  isMainnet: boolean,
+  selected: CellDepsSelected,
+): Promise<CKBComponents.CellDep[]> => {
+  let rgbppLockDep = getRgbppLockDep(isMainnet);
+  let btcTimeDep = getBtcTimeLockDep(isMainnet);
+  let xudtDep = getXudtDep(isMainnet);
+  let uniqueDep = getUniqueTypeDep(isMainnet);
+
+  const cellDepsObj = await fetchCellDepsJson();
+  if (cellDepsObj) {
+    rgbppLockDep = isMainnet ? cellDepsObj.rgbpp.mainnet : cellDepsObj.rgbpp.testnet;
+    btcTimeDep = isMainnet ? cellDepsObj.btcTime.mainnet : cellDepsObj.btcTime.testnet;
+    if (!isMainnet) {
+      xudtDep = cellDepsObj.xudt.testnet;
+      uniqueDep = cellDepsObj.unique.testnet;
+    }
+  }
+  let cellDeps: CKBComponents.CellDep[] = [];
+  if (selected.rgbpp) {
+    cellDeps = [
+      ...cellDeps,
+      rgbppLockDep,
```

Comment:
I don't understand why add an RGBPP lock dep with index `0x1`. Could you please add some comments to clarify?

Reply:
Added in https://github.com/ckb-cell/rgbpp-sdk/pull/191/commits/2730f59bd9aa7b2e09902e94d97ae3793fef541b